flink-kubernetes-operator/src/test/java/org/apache/flink/kubernetes/operator/FlinkOperatorTest.java

```diff
@@ -70,10 +70,21 @@ public void testConfigurationPassedToJOSDK() {
 
         var configService = testOperator.getOperator().getConfigurationService();
 
-        // Test parallelism being passed
+        // Test parallelism being passed expectedly
         var executorService = configService.getExecutorService();
         Assertions.assertInstanceOf(ThreadPoolExecutor.class, executorService);
         ThreadPoolExecutor threadPoolExecutor = (ThreadPoolExecutor) executorService;
+        for (int i = 0; i < testParallelism * 2; i++) {
+            threadPoolExecutor.execute(
+                    () -> {
+                        try {
+                            Thread.sleep(1000);
+                        } catch (InterruptedException e) {
+                            e.printStackTrace();
+                        }
+                    });
+        }
```

Comment:
what do we expect to test here?

Reply:
I want to test whether the number of threads in the executor can reach our expected maximumPoolSize when the executor receives enough tasks.
see assertion in line 87

Reply:
actually, I think code in lines 78 to 85 was ugly, any good suggestion?

Reply:
Maybe we could do a more functional test that will hold even if the service implementation changes. 

1. Set a smaller parallelism to not use soo many threads
2. For 2x the parallelism like you did we execute:
 - Increment atomic counter
 - Sleep for long (lets say 10sec)
 3. Assert the counter == 2x parallelism
 
 Wdyt? It would be good to test that the test actually fails before the change

Reply:
do you mean "increase counter and sleep for long in every task" ? if yes,  "Assert the counter == 2x parallelism" will always fail even if we set a fixed thread pool(what we want to change) 
I think check pool size (which equals really thread number）is functional enough.

Reply:
fine, by me for now